code/modules/admin/verbs/playsound.dm

```diff
@@ -52,84 +59,91 @@
 	playsound(get_turf(src.mob), S, 50, 0, 0)
 	SSblackbox.record_feedback("tally", "admin_verb", 1, "Play Local Sound") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
 
+/proc/invoke_youtubedl(url, dump_error_to_chat, mob/dump_to)
+	if(!istext(url))
+		return YTDL_FAILURE_NO_URL
+	var/ytdl = CONFIG_GET(string/invoke_youtubedl)
+	if(!ytdl)
+		return YTDL_NOT_CONFIGURED
+	url = trim(url)

```

Comment:
This shouldn't be done in the invoke_youtubedl proc, it should still be done in play_web_sound

Reply:
will there ever be a need for it to not be trimmed in this though?

Reply:
No, there isn't, but I think this proc should error if someone doesn't trim it before passing it to this proc

Reply:
what, no, this proc is so I can have players safely call YouTube dl without security risks

In fact this should verify it is a yt url before accessing

Reply:
There is no such thing as a player safe youtube dl call.

Reply:
we hashed out how we can do player safe youtube dl call in IRC

Reply:
That limits this so much, a proc for players to safely call youtube-dl should be a new, separate one.

Reply:
@JJRcop We can't trust clients with unrestricted shell access.

Reply:
@Iamgoofball The user-input handling shouldn't happen in this proc, it should happen in the proc that actually deals with user input, make another proc that calls this one and in that proc you handle the user input and trim it before passing it here, this isn't about sanitization, sanitization should definitely happen here, but I think the invoke_youtubedl proc should error out if the url isn't properly trimmed before being passed in, since the invoke_youtubedl proc is not user facing.